config/passport.js

```diff
@@ -0,0 +1,40 @@
+const passport = require('passport');
+const LocalStrategy = require('passport-local').Strategy;
+const User = require('../models/user');
+const bcrypt = require('bcrypt');
+
+
+passport.use(new LocalStrategy(
+  function(username, password, done) {
+    User.findOne({ username: { $regex: username, $options: 'i' }}, function (err, user) {
+      if (err) { return done(err); }
+      if (!user) {
+        return done(null, false, { message: 'Incorrect username' });
+      }
+
+      bcrypt.compare(password, user.password, function(err, isMatch) {
```

Comment:
Any specific reasons why you compare it here instead of doing it in the `model` instead?